docs/citycloud-snippets.md

```diff
@@ -0,0 +1,130 @@
+# Citycloud snippets
+
+To set up a cluster on citycloud these snippets can be used
+
+Start by setting up some environments for this setup
+
+```bash
+SERVICE_CLUSTER="my-new-sc-cluster"
+WORKLOAD_CLUSTERS=( "my-new-wc-cluster" )
+PUB_SSH_KEY_FILE="${HOME}/.ssh/id_rsa.pub"
+```
+
+> Tip:
+> Save these into a file `my-new-cluster.env`.
+> The next time you want to use these snippets you can just run `source my-new-cluster.env`
+
+Then you'll need to set up some openstack specific variables for the infrastructure
+
+```bash
+OS_IMAGE_NAME="Ubuntu 20.04 Focal Fossa 20200423" # Image name to base VMs on
+OS_NUM_MASTERS=3
+OS_MASTERS_FLAVOR="" # UUID of master flavor
+OS_NUM_WORKERS=3
+OS_WORKER_FLAVOR="" # UUID of worker flavor
+OS_EXT_NET_ID="" # UUID of the external network
+OS_SUBNET_CIDR="10.10.0.0/24" # (Guessing) CIDR of network
+OS_FLOATINGIP_POOL="" # Name of the pool to get floating ips from
+```
+
+Set up the clusters into respective folders
+
+```bash
+pushd kubespray
+for CLUSTER in "${SERVICE_CLUSTER}" "${WORKLOAD_CLUSTERS[@]}"; do
+  mkdir -p "inventory/${CLUSTER}/group_vars"
```

Comment:
I know this is where the openstack terraform code/docs thinks it should be, but I think it would be better if we could stick to the `$CK8S_CONFIG_PATH`.  My suggestion is to link `contrib/terraform/openstack` into `$CK8S_CONFIG_PATH` or  `$CK8S_CONFIG_PATH/$CLUSTER-config`. This way we keep the terraform state and `cluster.tfvars` under the config path so it is easier to keep track of it.

Reply:
I prefer this approach. It is closer aligned with kubespray. Terraform state should eventually be handled with `-backend-config`.

Reply:
Ok the state can be handled with a backend config, but what about `cluster.tfvars`? Surely we should keep that in the config folder?

Reply:
Kubeconfig encourages keeping `cluster.tfvars` in the `inventory/$CLUSTER` folder.

Reply:
Yes but I don't think it makes sense to have it outside of `$CK8S_CONFIG_PATH`. If we want it in the inventory to stick to the kubespray examples, then I think we should change the config path like this:

```bash
export CK8S_CONFIG_PATH="inventory/${CLUSTER}"
```

Reply:
I see your point. Let's go with your suggestion.